File: AppDistricon/backend/services/mercadopago.py
"""Mercado Pago API connector — mock by default."""
import requests
from flask import current_app

API_BASE = 'https://api.mercadopago.com'

def _headers():
    token = current_app.config.get('MERCADOPAGO_ACCESS_TOKEN', '')
    return {'Authorization': f'Bearer {token}'}

def get_payments(limit=25, offset=0):
    if current_app.config.get('MOCK_MERCADO_PAGO', True):
        return _mock_payments(limit)
    try:
        resp = requests.get(f'{API_BASE}/v1/payments/search', headers=_headers(),
                            params={'limit': limit, 'offset': offset})
        resp.raise_for_status()
        return resp.json().get('results', [])
    except Exception:
        return _mock_payments(limit)

def get_movements(limit=15):
    if current_app.config.get('MOCK_MERCADO_PAGO', True):
        return _mock_movements(limit)
    return _mock_movements(limit)

def get_balance():
    if current_app.config.get('MOCK_MERCADO_PAGO', True):
        return {'available_balance': 892340.00, 'pending_balance': 423500.00,
                'total_balance': 1315840.00}
    try:
        resp = requests.get(f'{API_BASE}/v1/account/money_balance', headers=_headers())
        resp.raise_for_status()
        return resp.json()
    except Exception:
        return {'available_balance': 892340.00, 'pending_balance': 423500.00, 'total_balance': 1315840.00}

def get_releases(limit=10):
    if current_app.config.get('MOCK_MERCADO_PAGO', True):
        return _mock_releases(limit)
    return _mock_releases(limit)

def _mock_payments(n):
    statuses = ['approved', 'pending', 'in_process', 'rejected', 'refunded']
    return [{'id': i, 'payment_id': 90000000+i, 'status': statuses[i%5],
             'amount': round(1500 + i*200, 2), 'net_received': round(1350 + i*180, 2),
             'date_created': f'2026-0{(i%9)+1:02d}-{(i%28)+1:02d}T12:00:00.000-03:00'}
            for i in range(1, n+1)]

def _mock_movements(n):
    types = ['income', 'expense', 'fee', 'refund']
    return [{'id': i, 'source_id': f'MOV{i}', 'type': types[i%4],
             'amount': round(200 + i*50, 2), 'balance': round(1000000 - i*250, 2),
             'date': f'2026-0{(i%9)+1:02d}-{(i%28)+1:02d}T10:00:00Z'} for i in range(1, n+1)]

def _mock_releases(n):
    statuses = ['available', 'pending', 'released']
    return [{'id': i, 'source_id': f'REL{i}', 'external_ref': f'REF{i:06d}',
             'release_date': f'2026-0{(i%9)+1:02d}-{(i%28)+1:02d}T10:00:00Z',
             'amount': round(5000 + i*300, 2), 'status': statuses[i%3]} for i in range(1, n+1)]

File: src/lib/mercadopago.ts
// Conector Mercado Pago Argentina
// Modo mock por defecto (MOCK_MERCADO_PAGO=true en .env)

import { getMercadoPagoCredentials } from '@/lib/mercadopago-config'

const MOCK_MODE = process.env.MOCK_MERCADO_PAGO !== 'false'

const API_BASE = 'https://api.mercadopago.com'

async function getAccessToken(): Promise<string | null> {
  if (MOCK_MODE) return 'mock_access_token'
  const credentials = await getMercadoPagoCredentials()
  return credentials?.accessToken || null
}

async function fetchMP(path: string) {
  const token = await getAccessToken()
  if (!token) return null
  const res = await fetch(`${API_BASE}${path}`, {
    headers: { Authorization: `Bearer ${token}` },
  })
  if (!res.ok) return null
  return res.json()
}

// ─── Mock Data ──────────────────────────────────────────

const MOCK_PAYMENTS = Array.from({ length: 25 }, (_, i) => ({
  id: 3000000 + i,
  order_id: 2309841 + i,
  status: ['approved', 'pending', 'rejected', 'refunded'][i % 4],
  payment_method: ['visa', 'master', 'rapipago', 'mercadopago_wallet'][i % 4],
  transaction_amount: 3000 + Math.round(Math.random() * 20000),
  net_amount: 2800 + Math.round(Math.random() * 18000),
  date_created: new Date(2026, 6, 5 - Math.floor(i / 4)).toISOString(),
  date_approved: new Date(2026, 6, 5 - Math.floor(i / 4) + 1).toISOString(),
}))

const MOCK_ACCOUNT_MOVEMENTS = Array.from({ length: 15 }, (_, i) => ({
  id: 5000000 + i,
  type: ['payment', 'refund', 'withdrawal', 'fee'][i % 4],
  status: ['available', 'unavailable', 'pending'][i % 3],
  amount: i % 2 === 0 ? 5000 + Math.round(Math.random() * 30000) : -2000 - Math.round(Math.random() * 10000),
  currency_id: 'ARS',
  date_created: new Date(2026, 6, 10 - i).toISOString(),
}))

// ─── Public API ──────────────────────────────────────────

export async function getPayments(fechaDesde?: string, fechaHasta?: string) {
  if (MOCK_MODE) return MOCK_PAYMENTS
  return fetchMP(`/v1/payments/search?sort=date_created&criteria=desc&range_date_created=${fechaDesde || '2026-01-01'},${fechaHasta || '2026-12-31'}`)
}

export async function getAccountMovements() {
  if (MOCK_MODE) return MOCK_ACCOUNT_MOVEMENTS
  return fetchMP(`/v1/advanced_payments/search`)
}

export async function getBalance() {
  if (MOCK_MODE) {
    return {
      total_available: 485000.50,
      total_pending: 156200.00,
      total_fees_current_month: 42300.80,
      total_refunds_current_month: 18300.25,
      total_releases_current_month: 312500.00,
      currency: 'ARS',
      release_summary: {
        by_week: [
          { week: 'Semana 1', amount: 82500 },
          { week: 'Semana 2', amount: 94500 },
          { week: 'Semana 3', amount: 71200 },
          { week: 'Semana 4', amount: 64300 },
        ],
      },
    }
  }
  return fetchMP(`/v1/account/money_balance`)
}

export async function getReleaseReport(fechaDesde?: string, fechaHasta?: string) {
  if (MOCK_MODE) {
    return {
      releases: Array.from({ length: 8 }, (_, i) => ({
        id: 6000000 + i,
        amount: 8000 + Math.round(Math.random() * 25000),
        status: ['released', 'pending', 'scheduled'][i % 3],
        payment_id: 3000000 + i * 3,
        release_date: new Date(2026, 6, 10 + i).toISOString(),
        estimated_release_date: new Date(2026, 6, 15 + i).toISOString(),
      })),
    }
  }
  return fetchMP(`/v1/releases/search?range_date=${fechaDesde || '2026-01-01'},${fechaHasta || '2026-12-31'}`)
}

export const isMock = () => MOCK_MODE
